app/controllers/admin/rewards_controller.rb

```diff
@@ -36,8 +36,12 @@ def update
     end
 
     def destroy
-      reward.destroy
-      redirect_to admin_rewards_url, notice: 'Reward was successfully destroyed.'
+      if reward.destroy
+        notice = 'Reward was successfully destroyed.'
+      else
+        notice = 'The reward cannot be deleted. It is assigned to the category.'
```

Comment:
I needed to add this restriction here so that the admin knows that he cannot remove the reward with the category.